android/app/src/main/java/app/organicmaps/editor/OsmLoginFragment.java

```diff
@@ -75,12 +95,63 @@ private void login()
     });
   }
 
+  private void loginWithBrowser()
+  {
+    enableInput(false);
+
+    String[] oauthParams = OsmOAuth.nativeOAuthParams();
+    Uri oauth2url = Uri.parse(oauthParams[0] + "/oauth2/authorize")
+            .buildUpon()
+            .appendQueryParameter("client_id", oauthParams[1])
+            .appendQueryParameter("scope", oauthParams[3])
+            .appendQueryParameter("redirect_uri", oauthParams[4])
+            .appendQueryParameter("response_type", "code")
+            .build();
+    try {
+      Intent myIntent = new Intent(Intent.ACTION_VIEW, oauth2url);
+      startActivity(myIntent);
+    } catch (ActivityNotFoundException e) {
+      //Toast.makeText(this, "No application can handle this request."
+      //        + " Please install a webbrowser",  Toast.LENGTH_LONG).show();
+      e.printStackTrace();
+    }
+  }
+
   private void enableInput(boolean enable)
   {
     mPasswordInput.setEnabled(enable);
     mLoginInput.setEnabled(enable);
     mLoginButton.setEnabled(enable);
     mLostPasswordButton.setEnabled(enable);
+    mLoginWebsiteButton.setEnabled(enable);
```

Comment:
It would be great to reuse the same login button (GP flavor uses web login, other flavors use password)